index.js

```diff
@@ -106,42 +104,44 @@ function forObject(appId, method, resource, body, callback) {
   });
 }
 
-function list(appId, resource, callback) {
-  return forList(appId, resource, callback);
+function list(context, appId, resource, callback) {
+  return forList(context, appId, resource, callback);
 }
 
-function get(appId, resource, callback) {
-  return forObject(appId, 'GET', resource, null, callback);
+function get(context, appId, resource, callback) {
+  return forObject(context, appId, 'GET', resource, null, callback);
 }
 
-function post(appId, resource, object, callback) {
-  return forObject(appId, 'POST', resource, object, callback);
+function post(context, appId, resource, object, callback) {
+  return forObject(context, appId, 'POST', resource, object, callback);
 }
 
-function put(appId, resource, object, callback) {
-  return forObject(appId, 'PUT', resource, object, callback);
+function put(context, appId, resource, object, callback) {
+  return forObject(context, appId, 'PUT', resource, object, callback);
 }
 
-function del(appId, resource, callback) {
-  return forObject(appId, 'DELETE', resource, null, callback);
+function del(context, appId, resource, callback) {
+  return forObject(context, appId, 'DELETE', resource, null, callback);
 }
 
 module.exports = (callerConfig) => {
-  config = callerConfig;
+  const config = callerConfig;
   if (!config || !config.apiUrl || !config.token || !config.secret) {
     throw new Error('You must provide apiUrl, token, and secret.');
   }
-  HmacAuthorize = require('authorization-hmac256')({
+  const hmacAuthorize = require('authorization-hmac256')({
     service: 'CWS',
     token: config.token,
     secret: config.secret
   });
-  return {
-    request,
-    list,
-    get,
-    post,
-    put,
-    del
+  const context = {config, hmacAuthorize};
+  const methods = {
+    request: (appId, method, resource, payload, pagination, callback) => request(context, appId, method, resource, payload, pagination, callback),
+    list: (appId, resource, callback) => list(context, appId, resource, callback),
+    get: (appId, resource, callback) => get(context, appId, resource, callback),
+    post: (appId, resource, object, callback) => post(context, appId, resource, object, callback),
+    put: (appId, resource, object, callback) => put(context, appId, resource, object, callback),
+    del: (appId, resource, callback) => del(context, appId, resource, callback)
```

Comment:
@brownjava  Can you update this and instead return these methods and maybe a new property on this object to retrieve the promised version of this?  This would break our lambda functionality since right now they are all callback based.